presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarController.java

```diff
@@ -0,0 +1,273 @@
+                    + " --config " + configDir.toAbsolutePath().resolve("config.properties"));
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to perform post install steps", e);
+        }
+    }
+
+    public void uninstallAsync(boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        if (!checkDependencies) {
+            throw new PrestoManagerException("Unsupported parameter 'checkDependencies' for tarball uninstall");
+        }
+        try {
+            deleteRecursively(INSTALLATION_DIR);
+            deleteRecursively(dataDir);
+            deleteRecursively(catalogDir);
+            deleteRecursively(configDir);
+            launcherScript = Optional.empty();
+            launcherConfig = Optional.empty();
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to uninstall Presto", e);
+        }
+        LOGGER.debug("Successfully uninstalled Presto");
+    }
+
+    public void upgradeAsync(URL packageUrl, boolean checkDependencies, boolean preserveConfig)
+            throws PrestoManagerException
+    {
+        if (!checkDependencies) {
+            throw new PrestoManagerException("Unsupported parameter 'checkDependencies' for tarball upgrade");
+        }
+        Path tempPackage = getTarPackage(packageUrl);
+        try {
+            if (preserveConfig) {
+                Path tempConfigDir = configDeployer.backupDirectory(configDir);
+                try {
+                    uninstallAsync(checkDependencies);
+                    tarInstall(tempPackage);
+                    postInstall();
+                    configDeployer.restoreDirectory(tempConfigDir, configDir);
+                }
+                finally {
+                    deleteTempFile(tempConfigDir);
+                }
+            }
+            else {
+                uninstallAsync(checkDependencies);
+                tarInstall(tempPackage);
+                postInstall();
+                configDeployer.deployDefaultConfig(configDir, dataDir, pluginDir);
+                configDeployer.deployDefaultConnectors(catalogDir);
+            }
+        }
+        finally {
+            deleteTempFile(tempPackage);
+        }
+        LOGGER.debug("Successfully upgraded presto");
+    }
+
+    public void startAsync()
+            throws PrestoManagerException
+    {
+        executeCommand("start");
+    }
+
+    public void terminate()
+            throws PrestoManagerException
+    {
+        executeCommand("stop");
+    }
+
+    public void kill()
+            throws PrestoManagerException
+    {
+        executeCommand("kill");
+    }
+
+    public void restartAsync()
+            throws PrestoManagerException
+    {
+        executeCommand("restart");
+    }
+
+    private int runLauncherCommand(String command)
+            throws PrestoManagerException
+    {
```

Comment:
No. this `runLauncherCommand` is basically a wrapper for running launcher commands. 
It runs the command like this:
```bin/launcher start CONFIGURATIONS```
CONFIGURATIONS is got from `launcherConfig`